pkg/nimble: add dependency to ble_nimble feature + remove board whitelist from tests and examples

aabadie opened the pull request:
<!--
The RIOT community cares a lot about code quality.
Therefore, before describing what your contribution is about, we would like
you to make sure that your modifications are compliant with the RIOT
coding conventions, see https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions.
-->

### Contribution description

This PR modifies the nimble applications to make use of `radio_nrfble` for nimble applications. Only nrf52 based boards are supported and provide this feature (until #11463 is merged).

It's simpler this way and more boards can be used (without an error message) with these applications.

<!--
Put here the description of your contribution:
- describe which part(s) of RIOT is (are) involved
- if it's a bug fix, describe the bug that it solves and how it is solved
- you can also give more information to reviewers about how to test your changes
-->


### Testing procedure

Build and test `examples/nimble_*` applications on nrf52 boards other than nrf52dk, nrf52840dk. They should just work fine.

<!--
Details steps to test your contribution:
- which test/example to compile for which board and is there a 'test' command
- how to know that it was not working/available in master
- the expected success test output
-->


### Issues/PRs references

None

<!--
Examples: Fixes #1234. See also #5678. Depends on PR #9876.

Please use keywords (e.g., fixes, resolve) with the links to the issues you
resolved, this way they will be automatically closed when your pull request
is merged. See https://help.github.com/articles/closing-issues-using-keywords/.
-->

aabadie commented:
I changed the initial strategy of this PR: now the `radio_nrfble` feature is a requirement of the nimble package (since it only supports nrf radio at the moment). This way there's no need for using a board whitelist in applications that use the nimble package.

haukepetersen commented:
+1 for this PR in general. However, I don't like using `radio_nrfble` as defining feature, as it describes our own, netdev based nrf-BLE-radio driver implementation, and this is completely unrelated to NimBLE.

I think it makes sense to introduce a new feature like `ble_nimble`. This feature can be defined by all `nrf`-based boards that meet the criteria e.g. not mapping the rtt and timer peripherals that NimBLE uses. Also we can add this feature later to boards, that e.g. define a UART/SPI HCI mapping for NimBLE to connect to an external BLE controller.

aabadie commented:
> I don't like using radio_nrfble as defining feature, as it describes our own, netdev based nrf-BLE-radio driver implementation, and this is completely unrelated to NimBLE.

In fact, radio_nrfble is unused in RIOT, that's why I reused this one. And also because I think it fits well here.

There is a `radio_ble` feature which is used by RIOT netdev based BLE radio driver implementation. Maybe it's the one you were thinking of ?

haukepetersen commented:
> Maybe it's the one you were thinking of ?

Nope, it seems to me that I did a mistake when defining the dependencies for `skald` -> it should require `radio_nrfble` instead of `radio_ble`.

But maybe this a good chance to clean up this mess altogether. So lets think about the features that we would need and what they tell us:
- `radio_nrfble` -> similar to `radio_nrfmin`, tells us that this platform can run the nrf-specific 'netdev/ble' radio driver. This is what `skald` should actually depend on. Further this should be moved to `cpu/nrf5x_common`, as it does not have any dependency into the board configuration I'd say.
- `radio_ble` -> not sure this generic feature a very useful. User applications would rather depend on a certain BLE stack implementation (`skald`, `nimble`, ...). And BLE stacks would rather depend on certain peripheral/radio/HCI configurations. So I tend to drop this generic feature for now.
- proposed `ble_nimble` -> tells us, that a certain platform has all needed configuration to run NimBLE. Alternatively we could generalize this to `ble_controller` and `ble_hci`?!

aabadie commented:
> lets think about the features that we would need and what they tell us

I have no strong opinion so I agree with what you propose. I updated the PR accordingly:
- radio_ble replaced by radio_nrfble in skald dependency
- dropped radio_ble
- add ble_nimble as feature provided by nrf52 boards and add it as a requirement for nimble package

It's all rebased and squashed.

haukepetersen commented:
I am happy and all checks pass -> go

aabadie commented:
Thanks for the review @haukepetersen !
